lib/src/mustachio/renderer_base.dart

```diff
@@ -145,7 +145,7 @@ abstract class RendererBase<T> {
   Template _template;
 
   /// The output buffer into which [context] is rendered, using a template.
-  final buffer = StringBuffer();
+  StringBuffer buffer = StringBuffer();
```

Comment:
Why can we not pass buffers into the RendererBase and share them, rather than requiring them to be different for each object?  Or, further, maybe this could be part of the idea @kevmoo suggested of just having a stream that we add to.

I'm OK with a big honking TODO/issue filed on cleaning this up, but I think the need for a `withBuffer` really points to a refactoring using streams being the way to handle this.

Reply:
I don't think there is any need for an asynchronous stream API. I've added a TODO to pass around a single StringBuffer.

Reply:
Oh, I was thinking StreamSink accessed synchronously.  Async definitely seems like overkill.

Reply:
Ah interesting.